landoapi/transplants.py

```diff
@@ -495,43 +606,330 @@ def check_landing_blockers(
+    revision_check_pairs = transplant_state.revision_check_pairs()
+
+    # Run revision-level blockers checks.
+    for revision, diff in revision_check_pairs:
+        phid = revision["phid"]
+        for block in revision_blockers:
+            reason = block(
+                revision=revision,
+                diff=diff,
+                transplant_state=transplant_state,
+            )
+            if reason is not None:
+                assessment.blockers.append(reason)
+
+                if phid is not None and phid in transplant_state.landable_stack:
+                    transplant_state.landable_stack.remove_node(phid)
+                    transplant_state.stack.nodes[phid]["blocked"].append(reason)
+
+    # Run revision-level warning checks.
+    for revision, diff in revision_check_pairs:
+        for check in revision_warnings:
+            reason = check(
+                revision=revision, diff=diff, transplant_state=transplant_state
+            )
+
+            if reason is not None:
+                assessment.warnings.append(reason)
+
+    return assessment
+
+
+def assess_transplant_request(
+    phab: PhabricatorClient,
+    supported_repos: dict[str, Repo],
     stack_data: RevisionData,
+    stack: RevisionStack,
+    relman_group_phid: str,
     data_policy_review_phid: str,
-):
-    """Build all transplant blocker checks that need extra Phabricator data"""
-    assert all((isinstance(r, Repo) for r in repositories.values()))
-
-    return DEFAULT_OTHER_BLOCKER_CHECKS + [
-        # Configure uplift check with extra data.
-        check_uplift_approval(relman_group_phid, repositories, stack_data),
-        # Configure data policy check with the project PHID.
-        check_revision_data_classification(data_policy_review_phid),
-    ]
+    landing_assessment: Optional[LandingAssessmentState] = None,
+) -> tuple[TransplantAssessment, TransplantAssessmentState]:
+    """Assess the transplant request."""
+    # Get data from Lando.
+    landable_repos = get_landable_repos_for_revision_data(stack_data, supported_repos)
+
+    involved_phids = set()
+    reviewers = {}
+    for revision in stack_data.revisions.values():
+        involved_phids.update(gather_involved_phids(revision))
+        reviewers[revision["phid"]] = get_collated_reviewers(revision)
+
+    # Get more Phabricator data.
+    involved_phids = list(involved_phids)
+    users = user_search(phab, involved_phids)
+    projects = project_search(phab, involved_phids)
+
+    secure_project_phid = get_secure_project_phid(phab)
+    testing_tag_project_phids = get_testing_tag_project_phids(phab)
+    testing_policy_phid = get_testing_policy_phid(phab)
+
+    # Create a copy of the stack where we will remove revisions that are blocked from
+    # landing, leaving a graph where each path is landable.
+    landable_stack = copy.deepcopy(stack)
+
+    # We only want to consider paths starting from the open revisions
+    # do grab the status for all revisions.
```

Comment:
This comment is unclear. Perhaps this also all belongs in `TransplantAssessmentState` instead. 

```suggestion
    # Grab the status for all revisions, since we only want to consider paths
    # starting from the open revision when assessing.
```